src/safeposix/syscalls/net_calls.rs

```diff
@@ -100,66 +118,66 @@ impl Cage {
     pub fn bind_syscall(&self, fd: i32, localaddr: &interface::GenSockaddr) -> i32 {
         self.bind_inner(fd, localaddr, false)
     }
+    fn bind_inner_socket(&self, sockfdobj: &mut SocketDesc, localaddr: &interface::GenSockaddr, prereserved: bool) -> i32 {
+        if localaddr.get_family() != sockfdobj.domain as u16 {
+            return syscall_error(Errno::EINVAL, "bind", "An address with an invalid family for the given domain was specified");
+        }
+        if sockfdobj.localaddr.is_some() {
+            return syscall_error(Errno::EINVAL, "bind", "The socket is already bound to an address");
+        }
+
+        let mut mutmetadata = NET_METADATA.write().unwrap();
+        let intent_to_rebind = sockfdobj.options & (1 << SO_REUSEPORT) != 0;
+
+        let newlocalport = if prereserved {
+            localaddr.port()
+        } else {
+            let localout = mutmetadata._reserve_localport(localaddr.addr(), localaddr.port(), sockfdobj.protocol, sockfdobj.domain, intent_to_rebind);
+            if let Err(errnum) = localout {return errnum;}
+            localout.unwrap()
+        };
+
+        let mut newsockaddr = localaddr.clone();
+        newsockaddr.set_port(newlocalport);
+
+        let sid = if let Some(id) = sockfdobj.socketobjectid {
+            id
+        } else {
+            let sock = interface::Socket::new(sockfdobj.domain, sockfdobj.socktype, sockfdobj.protocol);
+            let id = mutmetadata.insert_into_socketobjecttable(sock).unwrap();
+            sockfdobj.socketobjectid = Some(id);
+            id
+        } ;
+        let locksock = mutmetadata.socket_object_table.get(&sid).unwrap().clone();
+        let sockobj = locksock.read().unwrap();
+        let bindret = sockobj.bind(&newsockaddr);
+
+        if bindret < 0 {
+            match Errno::from_discriminant(interface::get_errno()) {
+                Ok(i) => {return syscall_error(i, "sendto", "The libc call to bind failed!");},
+                Err(()) => panic!("Unknown errno value from socket bind returned!"),
+            };
+        }
+
+        sockfdobj.localaddr = Some(newsockaddr);
+ 
+        0
+    }
```

Comment:
same

Reply:
This one further is also a factored out helper so that's why these are grouped.